### ISSUE TITLE
`function_body_length` error value in configuration file not detected if no warning provided

### ISSUE THREAD

**Issue by gchiacchio** [realm/SwiftLint#4645]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
```
swiftlint --version
0.50.3
```
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
Homebrew

* Paste your configuration file:

```yml
# insert yaml contents here
function_body_length:
    error: 4

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
```
Xcode 13.3.1
Build version 13E500a
```
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

THIS IS NOT TRIGGERING `function_body_length` (IT SHOULD)
```
echo "// Test
func test() {
    print("1")
    print("2")
    print("3")
    print("4")
    print("5")
}
" | swiftlint lint --no-cache --use-stdin

<nopath>:9:1: error: Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
Done linting! Found 1 violation, 1 serious in 1 file.
```
NOW IF I ADD A `warning` LINE, IT WORKS
```yml
# insert yaml contents here
function_body_length:
    warning: 4
    error: 4

```
```
echo "// Test
func test() {
    print("1")
    print("2")
    print("3")
    print("4")
    print("5")
}
" | swiftlint lint --no-cache --use-stdin

<nopath>:2:6: error: Function Body Length Violation: Function body should span 4 lines or less excluding comments and whitespace: currently spans 5 lines (function_body_length)
<nopath>:9:1: error: Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
Done linting! Found 2 violations, 2 serious in 1 file.
```

**Comment by gchiacchio** [comment]
note it's also failing for other rules as `type_body_length`, haven't tested others yet
